Lib/test/test_asyncio/test_waitfor.py

```diff
@@ -289,6 +288,42 @@ async def test_cancel_blocking_wait_for(self):
     async def test_cancel_wait_for(self):
         await self._test_cancel_wait_for(60.0)
 
+    async def test_wait_for_cancel_suppressed(self):
+        # See https://github.com/python/cpython/issues/84849
+        # See https://github.com/python/cpython/pull/28149
+
+        async def return_42():
+            try:
+                await asyncio.sleep(10)
+            except asyncio.CancelledError:
+                return 42
+
+        res = await asyncio.wait_for(return_42(), timeout=0.1)
+        self.assertEqual(res, 42)
```

Comment:
I still feel this should raise TimeoutError. We should probably discuss this back on GH-86296.

Reply:
Sorry, I take it back. It seems wait_for() currently works this way (at least since 3.9) and so does timeout().